rand_core/src/lib.rs

```diff
@@ -186,6 +186,19 @@ pub trait RngCore {
     /// 
     /// [`fill_bytes`]: trait.RngCore.html#method.fill_bytes
     fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error>;
+
+    /// Number of bytes generated per round of this RNG.
+    ///
+    /// Some algorithms would benefit from knowing some basic properties about
+    /// the RNG. In terms of performance an algorithm may want to know whether
+    /// an RNG is best at generating `u32`s, or could provide `u64`s or more at
+    /// little to no extra cost.
+    ///
+    /// For many RNGs a simple definition is: the smallest number of bytes this
+    /// RNG can generate without throwing away part of the generated value.
+    ///
+    /// `bytes_per_round` has a default implementation that returns `4` (bytes).
+    fn bytes_per_round(&self) -> usize { 4 }
```

Comment:
Could this be an associated constant?

Reply:
I tried something like that in a not really thought through attempt: https://github.com/rust-lang-nursery/rand/issues/377#issuecomment-379264500. The problem is that we then can't make `RngCore` into a trait object.

Reply:
Weird; doesn't sound like associated constants should prevent a trait from becoming object-safe.